### PR TITLE
first submmition

### PR DESCRIPTION
**Pull request by Roniana11**

### CODE REVIEW

File: src/ex2/ItemManager.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  }
+
+  removeAll(){
+    while (this.todos.length) {
+      this.todos.pop();
+    }
+  }
+
+  async addPokemons(ids) {
+    const pokemonsArray = ids.split(',');
```
Comment (barcohen2) [inline src/ex2/ItemManager.js]: ids should be an array and the conversion of user input (the split..) should be outside.
Item manager should not be aware of the user input format and should not handle it's parsing as well

File: src/ex2/PokemonClient.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+export class PokemonClient {
```
Comment (barcohen2) [inline src/ex2/PokemonClient.js:1]: Why do you need two methods in order to fetch pokemons? you can use only the fetchPokemons method and remove fetchSinglePokemon.
this way you don't have logic duplication and you can just send fetchPokemons an array that contains only one id.

File: src/ex2/PokemonClient.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  async fetchPokemons(ids) {
+    const allPromises = [];
+    ids.forEach((id) => {
+      allPromises.push(fetch(`https://pokeapi.co/api/v2/pokemon/${id}/`));
+    });
+
+    try {
+      let AllResponses = await Promise.all(allPromises);
+      AllResponses = [...AllResponses.map((response) => response.json())];
+      let AllData = await Promise.all(AllResponses);
```
Comment (barcohen2) [inline src/ex2/PokemonClient.js:13]: Im not sure I understand why do you need promise all here. All responses should be an array of the parsed responses.

File: src/ex2/index.html [hunk 4]
```diff
@@ -17,8 +17,13 @@ <h1 class="app-name">Todo App</h1>
...
         <ul id="list"></ul>
+        <div class="delete-all">
+        <button type="button" id="delete-all-button">Delete All</button>
+        </div>
       </div>
     </div>
   </body>
-  <script src="main.js"></script>
+  <script type="module" src="main.js"></script>
+  <script type="module" src="ItemManager.js"></script>
```
Comment (barcohen2) [inline src/ex2/index.html]: you dont have to import ItemManager.js and PokemonClient.js as they're imported in main.js

File: src/ex2/main.js [hunk 5]
```diff
@@ -1,10 +1,98 @@
...
+    );
+    listItem.addEventListener('click', () => this.onItemClick(listItem));
+
+    return listItem;
+  }
+
+  deleteItemHandler(todo, event) {
+    event.stopPropagation();
+    this.itemManager.removeItem(todo.id);
+    this.listEl.removeChild(todo);
```
Comment (barcohen2) [inline src/ex2/main.js]: Remove todos by updating the list and re-rendering

File: src/ex2/ItemManager.js (hunk 1 above)
Comment (Roniana11) [reply src/ex2/ItemManager.js]: fixed!

File: src/ex2/PokemonClient.js (hunk 2 above)
Comment (Roniana11) [reply src/ex2/PokemonClient.js:1]: fixed!

File: src/ex2/PokemonClient.js (hunk 3 above)
Comment (Roniana11) [reply src/ex2/PokemonClient.js:13]: then how can I resolve the response to get the pokemon object?

File: src/ex2/index.html (hunk 4 above)
Comment (Roniana11) [reply src/ex2/index.html]: fixed!

File: src/ex2/main.js (hunk 5 above)
Comment (Roniana11) [reply src/ex2/main.js]: fixed!

**Pull request closed without merging**